src/test/java/org/fundacionjala/coding/christian/CesarCryptTest.java

```diff
@@ -0,0 +1,187 @@
+     */
+    @Test
+    public void shouldCipherLetter() {
+        final int key1 = 0;
+        final int key2 = 1;
+        final int key3 = 3;
+        final int key4 = 5;
+        final int key5 = -3;
+        final int key6 = -10;
+        assertEquals("A", cesar.encode("A", key1));
+        assertEquals("B", cesar.encode("A", key2));
+        assertEquals("D", cesar.encode("A", key3));
+        assertEquals("J", cesar.encode("E", key4));
+        assertEquals("A", cesar.encode("Z", key2));
+        assertEquals("C", cesar.encode("X", key4));
+        assertEquals("Z", cesar.encode("C", key5));
+        assertEquals("X", cesar.encode("H", key6));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void shouldCipherWholeAlphabet() {
+        final int key = -3;
+        String allChars = "THE QUICK BROWN FOX JUMPS OVER THE LAZY DOG";
+        assertEquals("QEB NRFZH YOLTK CLU GRJMP LSBO QEB IXWV ALD",
+                cesar.encode(allChars, key));
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void shouldDecipher() {
+        final int key1 = 2;
+        final int key2 = -2;
+        String s = "aoeu";
+        String encrypted = cesar.encode(s, key1);
+        assertEquals(s, cesar.encode(encrypted, key2));
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void testEncodeWithPositiveKey() {
+        final int key1 = 3;
+        final int key2 = 5;
+        assertEquals("KROD", cesar.encode("HOLA", key1));
+        assertEquals("ETWWT", cesar.encode("ZORRO", key2));
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void testEncodeWithNegativeKey() {
+        final int key = -3;
+        assertEquals("HOLA", cesar.encode("KROD", key));
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void testEncodeWithEmptyMessage() {
+        final int key = 5;
+        assertEquals("", cesar.encode("", key));
+        assertEquals("", cesar.encode("", key));
+
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void testEncodeWithNullMessage() {
+        final int key = 5;
+        assertEquals("", cesar.encode(null, key));
+        assertEquals("", cesar.encode(null, key));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testEncodeWithSpecialCharactersAndBlankSpace() {
+        final int key = 3;
+        assertEquals("K_ROD", cesar.encode("H_OLA", key));
+        assertEquals("KROD CRUUR", cesar.encode("HOLA ZORRO", key));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testDecodeWithPositiveKey() {
+        final int key = 3;
+        assertEquals("A", cesar.encode("X", key));
+        assertEquals("Z", cesar.encode("W", key));
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void testDecodeWithNegativeKey() {
+        final int key = -3;
+        assertEquals("ZEPA", cesar.encode("CHSD", key));
+        assertEquals("WWACP PAC", cesar.encode("ZZDFS SDF", key));
+    }
+
+
+    /**
+     *
+     */
+    @Test
+    public void testDecodeWithEmptyMessage() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "shouldDoNothingWithEmptyString" on line 29. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)